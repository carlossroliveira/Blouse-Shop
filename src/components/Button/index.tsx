// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, ButtonSC } from './buttonStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonProps } from './types';

export const Button = (props: ButtonProps) => {
  return (
    <ContainerSC>
      <ButtonSC role="button">{props.text}</ButtonSC>
    </ContainerSC>
  );
};
