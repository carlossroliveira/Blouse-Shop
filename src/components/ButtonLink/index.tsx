// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, LinkSC } from './buttonLinkStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonLinkProps } from './types';

export const ButtonLink = (props: ButtonLinkProps) => {
  return (
    <ContainerSC>
      <LinkSC to={`${props.path}`} onClick={props.onClick}>
        {props.text}
      </LinkSC>
    </ContainerSC>
  );
};
