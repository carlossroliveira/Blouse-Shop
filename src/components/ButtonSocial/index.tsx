// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ButtonSC } from './buttonSocialStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonSocialProps } from './types';

export const ButtonSocial = (props: ButtonSocialProps) => {
  return (
    <ButtonSC role="button">
      <>
        {props.icon}
        {props.text}
      </>
    </ButtonSC>
  );
};
