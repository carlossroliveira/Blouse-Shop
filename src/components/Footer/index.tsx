// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, ParagraphSC } from './footerStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { IFooterProps } from './types';

export const Footer = (props: IFooterProps) => {
  const currentYear: number = new Date().getFullYear();

  return (
    <ContainerSC>
      <ParagraphSC Color={props.color}>
        &copy; {currentYear} | Carlos Oliveira
      </ParagraphSC>
    </ContainerSC>
  );
};
