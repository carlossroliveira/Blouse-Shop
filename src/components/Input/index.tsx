// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './inputStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { InputProps } from './types';

export const Input = (props: InputProps) => {
  return (
    <ContainerSC>
      <input
        type={props.type}
        value={props.value}
        onChange={props.onChange}
        placeholder={props.placeholder}
      />
    </ContainerSC>
  );
};
