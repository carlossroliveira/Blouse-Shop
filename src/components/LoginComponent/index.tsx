import React, { FormEvent, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { useAuth } from '../../hooks/useAuth';

export const LoginComponent = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const NAVIGATION = useNavigate();
  const auth = useAuth();

  const handleSubmita = async (event: FormEvent) => {
    event.preventDefault();

    console.log(email);
    console.log(password);
    try {
      await auth.authenticate(email, password);

      NAVIGATION('/profile');
    } catch (err) {
      alert(err);
    }
  };

  /*  async function onFinish(event: any, email: string, password: string) {
    event.preventDefault();
    try {
      await auth.authenticate(email, password);

      NAVIGATION('/profile');
    } catch (err) {
      alert(err);
    }
  } */

  return (
    <>
      <form onSubmit={handleSubmita}>
        <div>
          <input
            type="text"
            placeholder="email"
            value={email}
            onChange={(event) => setEmail(event.target.value)}
          />
        </div>
        <div>
          <input
            type="text"
            placeholder="Senha"
            value={password}
            onChange={(event) => setPassword(event.target.value)}
          />
        </div>
        <button type="submit">Logar</button>
      </form>
    </>
  );
};
