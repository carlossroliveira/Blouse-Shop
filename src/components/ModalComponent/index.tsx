// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { useState } from 'react';
import Modal from 'react-modal';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './modalStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

/* const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
  },
}; */

export const ModalComponent = () => {
  const [modalIsOpen, setIsOpen] = useState<boolean>(false);

  const openModal = () => setIsOpen(true);

  const closeModal = () => setIsOpen(false);

  return (
    <ContainerSC>
      <button onClick={openModal}>Open Modal</button>

      <Modal
        isOpen={modalIsOpen}
        onRequestClose={closeModal}
        /* style={customStyles} */
        contentLabel="Example Modal"
      >
        aaaaaaaaaaaaaa
      </Modal>
    </ContainerSC>
  );
};
