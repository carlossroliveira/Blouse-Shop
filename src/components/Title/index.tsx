// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, TitleSC } from './titleStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ITitleProps } from './types';

export const Title = (props: ITitleProps) => {
  return (
    <ContainerSC>
      <TitleSC>{props.title}</TitleSC>
    </ContainerSC>
  );
};
