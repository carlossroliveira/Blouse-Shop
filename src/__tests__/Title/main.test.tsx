// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { Title } from '../../components/Title';
// -------------------------------------------------
// Utils
// -------------------------------------------------
import WithThemeComponent from '../../utils/withThemeComponent';

const LoginWithTheme = WithThemeComponent(Title);

describe('Testing error Title', () => {
  it('Should render the title correctly', () => {
    render(<LoginWithTheme title={'Testing'} />);

    expect(
      screen.getByRole('heading', {
        name: /testing/i,
      }),
    ).toBeInTheDocument();
  });

  it('component rendering matches snapshot correctly', () => {
    render(<LoginWithTheme title={'Testing'} />);
    expect(render(<LoginWithTheme title={'Testing'} />)).toMatchSnapshot();
  });
});
