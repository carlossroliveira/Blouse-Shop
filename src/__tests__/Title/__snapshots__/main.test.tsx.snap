// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing error buyBlouse component rendering matches snapshot correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  color: #0000;
  overflow: hidden;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  font-size: 2.5rem;
  line-height: 1.2em;
  text-shadow: 0 0 #383838,0 1.2em #383838;
}

.c0:hover {
  text-shadow: 0 -1.2em #383838,0 0 #383838;
}

<body>
    <div>
      <section
        class=""
      >
        <h1
          class="c0"
        >
          Testing
        </h1>
      </section>
    </div>
    <div>
      <section
        class=""
      >
        <h1
          class="c0"
        >
          Testing
        </h1>
      </section>
    </div>
  </body>,
  "container": .c0 {
  color: #0000;
  overflow: hidden;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  font-size: 2.5rem;
  line-height: 1.2em;
  text-shadow: 0 0 #383838,0 1.2em #383838;
}

.c0:hover {
  text-shadow: 0 -1.2em #383838,0 0 #383838;
}

<div>
    <section
      class=""
    >
      <h1
        class="c0"
      >
        Testing
      </h1>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
